### PR TITLE
Change amount field to support with decimals

### PR DESCRIPTION
**Pull request by youngkidwarrior**

### CODE REVIEW

**Comment by youngkidwarrior** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/443?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#443** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/443?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#442** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/442?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#441** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/441?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#439](https://github.com/0xsend/sendapp/pull/439) <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/439?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#440** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/440?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @youngkidwarrior and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/app/features/send/SendAmountForm.tsx [hunk 1]
```diff
@@ -115,7 +116,10 @@ export function SendAmountForm() {
             fontFamily: '$mono',
             keyboardType: balance?.decimals ? 'decimal-pad' : 'numeric',
             onChangeText: (text) => {
-              form.setValue('amount', text.replace(/[^0-9.]/g, ''))
+              const [beforeDecimal, ...afterDecimal] = text.split('.')
+              const formattedText =
+                beforeDecimal + (afterDecimal.length ? `. + ${afterDecimal.join('')}` : '')
```
Comment (0xBigBoss) [inline packages/app/features/send/SendAmountForm.tsx:121]: confused at the interpolated string is trying to do. 

``. + ${afterDecimal.join('')}`` 

Did you mean `.${afterDecimal.join('')}`

Comment (youngkidwarrior) [reply packages/app/features/send/SendAmountForm.tsx]: Yeah this was super unclear. Thanks for calling me out. I refactored and added a note to clear up any magic

Comment (0xBigBoss) [reply packages/app/features/send/SendAmountForm.tsx]: i got that part. I'm thinking the [string interpolation is unnecessary](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBATgUwLYgG4ICIFcAOAbAS2AEMoEBJMAZSjgLAHMYBeGACjIA8oAuGaOowA0-LACMB9Bn0mMAlCwB8MAN4AoGDFCRYAbQBmBONBEA6c8X1k4AXRYwuUUxHwEobCONkM5G+AigsODAYQ2NYAGp2S2tTPARGKAALGAB+GAADABIVTwlaKQBfGCicmIQ4UwArEHo2AHJ6uUKMmD5G300Aei7-QODQo2gS6KsKuISGZLTMnLzvQrKxypq6jpa2mA61QrU1bWHEFHRsVxIyShpBBgA5EEprHHsOBG4ZAuFRfOv364VmZTqTQHPRhEwwcymcq2eyOZyudzzD4+PyIfohMGRUaxeKJFLpbK5LzIxYqaHVWpgBpNDbtJo7PYHEDxOIgBhsI5oTC4QjnCjUZENACMACYAMwQmCiiWmKXi+oieqmJq+JksvBsjnILmnXmkflXKR3B4VHDC8WS6WW+WK5VyORAA).  Why add the `+` to the string?

Screenshot 2024-06-06 at 09 52 00

Comment (youngkidwarrior) [reply packages/app/features/send/SendAmountForm.tsx]: Ah yeah that's a mistake

**Review by 0xBigBoss — APPROVED** [review]

**Comment by youngkidwarrior** [comment]
### Merge activity

* **Jun 9, 1:17 AM EDT**: @youngkidwarrior started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/0xsend/sendapp/443).

**Pull request merged**